client/gutenberg/editor/controller.js

```diff
@@ -94,13 +101,19 @@ export const authenticate = ( context, next ) => {
 	const state = context.store.getState();
 
 	const siteId = getSelectedSiteId( state );
+	const isDesktop = isEnabled( 'desktop' );
 	const storageKey = `gutenframe_${ siteId }_is_authenticated`;
 
-	const isAuthenticated =
+	let isAuthenticated =
 		globalThis.sessionStorage.getItem( storageKey ) || // Previously authenticated.
 		! isSiteWpcomAtomic( state, siteId ) || // Simple sites users are always authenticated.
-		isEnabled( 'desktop' ) || // The desktop app can store third-party cookies.
+		isDesktop || // The desktop app can store third-party cookies.
 		context.query.authWpAdmin; // Redirect back from the WP Admin login page to Calypso.
+
+	if ( isDesktop && isJetpackSite( state, siteId ) && ! ssoEnabled( reduxGetState(), siteId ) ) {
```

Comment:
Note: will remove the `reduxGetState` here and just use `state`. Was seeing if this made a difference to the "prompted twice" issue (it doesn't). 😕 

TL;DR - Combined with changes in the desktop, this logic works as expected, but for some reason `sso` status read from the store is stale after the first attempt, and only becomes correct after the second prompt to the user. I'm using the `requestSite` call to update the site's state - perhaps my understanding of what this method does is incorrect.

Reply:
Going to remove `reduxGetState()` and replace with `state`